src/pages/BlockDetail/BlockComp.tsx

```diff
@@ -90,7 +90,12 @@ const BlockMinerReward = ({
   )
 }
 
-export const BlockOverviewCard: FC<{ blockHeightOrHash: string; block: Block }> = ({ blockHeightOrHash, block }) => {
+export const BlockOverviewCard: FC<{
+  blockHeightOrHash: string
+  block: Block
+  blockNumber: number
+  switchBlockNumber?: (step: number) => number
```

Comment:
The name `switchBlockNumber` seems more like an action rather than a getter.

Consider renaming it to `getRelativeBlockNumber` or `getNextBlockNumber`. Otherwise, both the places where the component is used and the internal implementation of the component may confuse developers.

Reply:
> The name `switchBlockNumber` seems more like an action rather than a getter.
> 
> Consider renaming it to `getRelativeBlockNumber` or `getNextBlockNumber`. Otherwise, both the places where the component is used and the internal implementation of the component may confuse developers.

`getNextBlockNumber` maybe confuse developers it only returns the `next` block, not `previous`

I don't know what the `getRelativeBlockNumber` could do when I saw this name

Reply:
> `getNextBlockNumber` maybe confuse developers it only returns the `next` block, not `previous`
> 
> I don't know what the `getRelativeBlockNumber` could do when I saw this name

I think `getRelativeBlockNumber` already expresses the intention of `getting` a specific `blockNumber` quite clearly. It is at least more meaningful than `switchBlockNumber`.

Reply:
I see that the current design has both `blockNumber` and `switchBlockNumber` properties.

We can consider using only one of them since the meaning of `blockNumber` is already similar to `switchBlockNumber`.

By adjusting the default value from `0` to `null`, they will be exactly the same.

Reply:
> I see that the current design has both `blockNumber` and `switchBlockNumber` properties.
> 
> We can consider using only one of them since the meaning of `blockNumber` is already similar to `switchBlockNumber`.
> 
> By adjusting the default value from `0` to `null`, they will be exactly the same.

Not the same.
We need to show the blockNumber from the blockHeight which is provided in the path parameter until the data query finished. And also the block number should be displayed to 0 when the default block number is set to null or undefined

Reply:
I may lost the context. The problem to solve is `next/previous block goes to 0 on page loading`? then how about disabling the button during the loading, or cache the block number during the loading?

BTW, for the naming of `switchBlockNumber`, `getRelativeBlockNumber`, `getNextBlockNumber`, I can give another one `getSiblingBlock(step: 1 | -1): number`

Reply:
Any update on this conversation

Reply:
> > And also the block number should be displayed to 0 when the default block number is set to null or undefined
> 
> It sounds like we just need to use `blockNumber ?? 0` at the final rendering place.

The type of this parameter is `number`, so I think it should only accept 0 or other number.

Reply:
> I may lost the context. The problem to solve is `next/previous block goes to 0 on page loading`? then how about disabling the button during the loading, or cache the block number during the loading?
> 
> BTW, for the naming of `switchBlockNumber`, `getRelativeBlockNumber`, `getNextBlockNumber`, I can give another one `getSiblingBlock(step: 1 | -1): number`

I think it is more easy to understand when I see the `switchBlockNumber`, because it tells me that the block number will be switch, and the step will be passed as the parameter. When I see the `getSiblingBlock`, the first thing I do is to find the meaning of "sibling block" by Google.

Reply:
> > I may lost the context. The problem to solve is `next/previous block goes to 0 on page loading`? then how about disabling the button during the loading, or cache the block number during the loading?
> > BTW, for the naming of `switchBlockNumber`, `getRelativeBlockNumber`, `getNextBlockNumber`, I can give another one `getSiblingBlock(step: 1 | -1): number`
> 
> I think it is more easy to understand when I see the `switchBlockNumber`, because it tells me that the block number will be switched, and the step will be passed as the parameter. When I see the `getSiblingBlock`, the first thing I do is to find the meaning of "sibling block" by Google.

In terms of naming, I will guess `switchBlockNumber` is a function of action, calling it navigates to the specified page(e.g. `switchLocale` to update the language, `switchTab` to activate a different tab), but it works as a getter, simply returns a string.

Reply:
How about navigating to the specific block page on calling it

Reply:
> How about navigating to the specific block page on calling it

I think maybe it is not suitable to define a function here, just use the `blockNumber + 1` and `blockNumber - 1` directly, and compute the disabled status directly is much more better. For example.

```typescript
<Link to={`/block/${blockNumber > 0 ? blockNumber - 1 : 0`} disabled={!ableToSwitchToPrevious} />
<Link to={`/block/${blockNumber + 1}`} disabled={!ableToSwitchToNext} />
```

It's more clearly.

Reply:
The original reason is that `blockNumber` will ephemerally be `0`, so `ableToSwitchPrevious` and `ableToSwitchToNext` should be `false` during loading

Reply:
> The original reason is that `blockNumber` will ephemerally be `0`, so `ableToSwitchPrevious` and `ableToSwitchToNext` should be `false` during loading

Now the blockNumber will not be the default value when switch the block, so it could be `true` during loading, it will lead to the next block or previous (if larger than zero).